plugins/modules/lxd_storage_volume_info.py

```diff
@@ -0,0 +1,431 @@
+                )
+
+        return resp_json.get("metadata", {})
+
+    def get_storage_volumes(self) -> list[dict]:
+        """Retrieve storage volume information based on module parameters."""
+        # First check if the pool exists
+        self._check_pool_exists()
+
+        storage_volumes = []
+
+        if self.name:
+            # Get information about a specific volume
+            # We need to determine the type if not specified
+            if self.volume_type:
+                volume_info = self._get_volume_info(self.volume_type, self.name)
+                storage_volumes.append(volume_info)
+            else:
+                # Try to find the volume by iterating through all volumes
+                volume_urls = self._get_volume_list()
+                found = False
+                for volume_url in volume_urls:
+                    vol_type, vol_name = self._parse_volume_url(volume_url)
+                    if vol_name == self.name:
+                        volume_info = self._get_volume_info(vol_type, vol_name)
+                        storage_volumes.append(volume_info)
+                        found = True
+                        break
+
+                if not found:
+                    self.module.fail_json(msg=f'Storage volume "{self.name}" not found in pool "{self.pool}"')
+        else:
+            # Get information about all volumes in the pool
+            volume_urls = self._get_volume_list()
+            for volume_url in volume_urls:
+                vol_type, vol_name = self._parse_volume_url(volume_url)
+                # Apply type filter if specified
+                if self.volume_type and vol_type != self.volume_type:
+                    continue
+                volume_info = self._get_volume_info(vol_type, vol_name)
+                storage_volumes.append(volume_info)
+
+        return storage_volumes
+
+    def run(self) -> None:
+        """Run the main method."""
+        try:
+            if self.trust_password is not None:
+                self.client.authenticate(self.trust_password)
+
+            storage_volumes = self.get_storage_volumes()
+
+            result_json = {"storage_volumes": storage_volumes}
+            if self.client.debug:
+                result_json["logs"] = self.client.logs
+            self.module.exit_json(**result_json)
+        except LXDClientException as e:
+            self._fail_from_lxd_exception(e)
+
+
+def main() -> None:
+    """Ansible Main module."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            pool=dict(type="str", required=True),
+            name=dict(type="str"),
+            type=dict(type="str"),
+            project=dict(type="str"),
+            url=dict(type="str", default=ANSIBLE_LXD_DEFAULT_URL),
+            snap_url=dict(type="str", default="unix:/var/snap/lxd/common/lxd/unix.socket"),
```

Comment:
As commented before, the snap URL should be a constant by the top of the file, close to ANSIBLE_LXD_DEFAULT_URL.

This now is not mu consistent.

As a matter of fact, the two constants should rather be in the module_utils file and be imported over here, in both modules.